docs/topics/downloader-middleware.rst

```diff
@@ -725,9 +725,11 @@ RedirectMiddleware
    This middleware handles redirection of requests based on response status.
 
 .. reqmeta:: redirect_urls
+.. reqmeta:: redirect_reasons
 
-The urls which the request goes through (while being redirected) can be found
-in the ``redirect_urls`` :attr:`Request.meta <scrapy.http.Request.meta>` key.
+The urls which the request goes through (while being redirected) and their 
+status or meta-refresh html tag can be found in the ``redirect_urls`` and 
+``redirect_reasons`` :attr:`Request.meta <scrapy.http.Request.meta>` keys respectively.
```

Comment:
Even after the latest changes, I don’t think it is clear what the contents of `redirect_reasons` are. Are HTTP status codes provided as integers or as strings? What’s the format of the ‘meta-refresh HTML tag’, is it provided as a string containing the whole tag, a selector or something else?